gui_app.py

```diff
@@ -496,6 +495,33 @@ def run_installer(self, evt=None):
     def reinstall_deps(self, evt=None):
         utils.check_dependencies()
 
+    def set_file_dialog_environment(self):
+        desktop_environment = os.environ.get('XDG_CURRENT_DESKTOP', '').lower()
+    
+        if 'gnome' in desktop_environment:
+            os.environ['TK_SILENCE_DEPRECATION'] = '1'
+            os.environ['GTK_FILE_CHOOSER_ACTION'] = 'open'
+        elif 'kde' in desktop_environment:
+            os.environ['TK_SILENCE_DEPRECATION'] = '1'
+            os.environ['KDE_FULL_SESSION'] = 'true'
+
+    def open_file_dialog(self, filetype_name, filetype_extension):
+        root = Tk()
+        root.withdraw()
+    
+        file_path = fd.askopenfilename(
+            title=f"Select {filetype_name}",
+            filetypes=[(filetype_name, f"*.{filetype_extension}"), ("All Files", "*.*")],
+        )
+
+        return file_path
+
+    def set_appimage(self, evt=None):
+        Tk().withdraw()
+        self.set_file_dialog_environment()
+        appimage_filename = self.open_file_dialog("AppImage", "AppImage")
```

Comment:
If the user cancels the dialog, it returns an empty tuple `()`. Maybe the best way to handle that is to add something like:
```python
if not appimage_filename:
    return
```